apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/registry/DatabaseDiscoveryClientImpl.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.biz.registry;
+
+import com.ctrip.framework.apollo.biz.entity.Registry;
+import com.ctrip.framework.apollo.biz.registry.configuration.support.ApolloRegistryDiscoveryProperties;
+import com.ctrip.framework.apollo.biz.registry.configuration.support.ApolloRegistryClientProperties;
+import com.ctrip.framework.apollo.biz.service.RegistryService;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatabaseDiscoveryClientImpl implements DatabaseDiscoveryClient {
+
+  private static final Logger log = LoggerFactory.getLogger(DatabaseDiscoveryClientImpl.class);
+
+  private final RegistryService registryService;
+
+  private final ApolloRegistryDiscoveryProperties discoveryProperties;
+
+  private final ServiceInstance self;
+
+  public DatabaseDiscoveryClientImpl(
+      RegistryService registryService,
+      ApolloRegistryDiscoveryProperties discoveryProperties,
+      ServiceInstance self) {
+    this.registryService = registryService;
+    this.discoveryProperties = discoveryProperties;
+    this.self = self;
+  }
+
+  /**
+   * find by {@link ApolloRegistryClientProperties#getServiceName()}
+   */
+  @Override
+  public List<ServiceInstance> getInstances(String serviceName) {
+    final List<Registry> filterByLabel;
+    {
+      List<Registry> all = this.registryService.findByServiceName(serviceName);
+      if (this.discoveryProperties.isFilterByLabel()) {
```

Comment:
Maybe portal is deployed in a special network zone, i.e it can access all service, so it want to discover all service in different network zone.

It is more simpler if we keep filtering by label as default policy, and delete this config, if someone want to discover all in the feature, we can add it back.

Reply:
> Maybe portal is deployed in a special network zone, i.e it can access all service, so it want to discover all service in different network zone.

There is a related issue #3638. Though we could implement this feature by not filtering by label, I think this is not the best approach, as this only works for database service discovery. As was discussed in the original issue, I think we may add a mechanism to config meta server by env+cluster.

> It is more simpler if we keep filtering by label as default policy, and delete this config, if someone want to discover all in the feature, we can add it back.

+1 for this approach, as the label is only used in filtering, the user could achieve the `discover all` feature by simply not configuring the label.